### PR TITLE
 Use auth-helper to auth webapp&functions.

### PR DESCRIPTION
**Pull request by andxu**
Use auth-helper to auth webapp&functions.

### CODE REVIEW

File: azure-spring-cloud-maven-common/src/test/resources/pom-1.xml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (Flanker32) [inline azure-spring-cloud-maven-common/src/test/resources/pom-1.xml]: Why we move the resources to `spring-cloud-maven-common` ?

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java [hunk 2]
```diff
@@ -250,8 +280,50 @@ protected void initTelemetry() {
...
+            } else {
+                throw new AzureAuthFailureException(String.format("Unable to get server('%s') from settings.xml.", auth.getServerId()));
+            }
+        }
+
+        final List<ConfigurationProblem> problems = auth.validate();
+        if (problems.stream().anyMatch(problem -> problem.getSeverity() == Severity.ERROR)) {
+            throw new AzureAuthFailureException(String.format("Unable to validate auth configuration due to the following errors: %s",
+                    problems.stream().map(ConfigurationProblem::getErrorMessage).collect(Collectors.joining("\n"))));
+        }
```
Comment (Flanker32) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java]: Shall we move the check of server id as well as the validate to ` MavenSettingHelper.getAuthConfigurationFromServer`

Comment (andxu) [reply azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/AbstractAzureMojo.java]: Should be done later

File: azure-spring-cloud-maven-common/src/test/resources/pom-1.xml (hunk 1 above)
Comment (andxu) [reply azure-spring-cloud-maven-common/src/test/resources/pom-1.xml]: Done

**Pull request merged**